examples/canvas/src/main.rs

```diff
@@ -0,0 +1,70 @@
+#[macro_use]
+extern crate stdweb;
+
+use stdweb::Reference;
+use stdweb::unstable::TryInto;
+use stdweb::web::{
+    IEventTarget,
+    IHtmlElement,
+    document,
+    window,
+    RenderingContext
+};
+
+use stdweb::web::event::{
+    IMouseEvent,
+    MouseMoveEvent,
+    ResizeEvent,
+};
+
+use stdweb::web::html_element::CanvasElement;
+
+// Shamelessly stolen from webplatform's TodoMVC example.
+macro_rules! enclose {
+    ( ($( $x:ident ),*) $y:expr ) => {
+        {
+            $(let $x = $x.clone();)*
+            $y
+        }
+    };
+}
+
+#[derive(Clone, Debug)]
+struct CanvasRenderingContext2D(Reference);
```

Comment:
As I wrote in one of the comments, I would be fine with having the `RenderingContext`s implementations inside of `stdweb`, and I think I'd actually prefer that right now, so please move this from the example to inside the `stdweb`.

Also, please rename the `2D` to `2d` to keep with the `PascalCase` naming convention we've decided the other day to stick to.

Reply:
Hm, OK I will move it in - I'm not planning on expanding on the set of implemented methods though.

I do think these would work better as a separate crate: have you thought about splitting up stdweb?

eg.
- stdweb-core (contains just webcore)
- stdweb-common (contains bindings to core js APIs, not specific to environment)
- stdweb-browser (browser-specific bindings)
- stdweb-node (node-specific bindings)

Adding *everything* into `stdweb` will increase compile times quite a bit, especially if you add the whole of canvas 2d, webgl and webgl 2

Reply:
Yeah, it's fine not adding every method there is right now.

Yes, I did think about splitting it up in the future! The first thing I'd like to split up is `webapi` and `webcore` directories, and then we'll go from there and see if/what would make sense to further split up. However, I don't want to do that just yet.

As far as the compile times go there is probably some room for improvement too without splitting it up. (E.g. once we make the `js!` macro into a procedural macro)